katsdpimager/preprocess.cpp

```diff
@@ -378,13 +354,38 @@ void visibility_collector<P>::compress(int channel)
         else
         {
             // Moved to the next output visibility
+            counts[last.w_slice]++;
             buffer[out_pos++] = last;
             last = element;
         }
     }
-    // Emit the final batch
+    // Write the last output visibility
+    counts[last.w_slice]++;
     buffer[out_pos++] = last;
-    emit(channel, buffer, out_pos);
+
+    // Prefix sum counts (in-place)
+    std::size_t sum = 0;
+    for (std::size_t &c : counts)
```

Comment:
It took me a while to grasp this loop... I like the fancy new (to me) syntax too :-)

Reply:
It's only 10 years old (C++11) :-)

Reply:
My main language switched from C++ to Python 15 years ago, so that would explain it...